src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeAugments.cs

```diff
@@ -143,42 +143,16 @@ public static unsafe Array NewMultiDimArray(RuntimeTypeHandle typeHandleForArray
             {
                 // We just checked above that all lower bounds are zero. In that case, we should actually allocate
                 // a new SzArray instead.
-                RuntimeTypeHandle elementTypeHandle = new RuntimeTypeHandle(typeHandleForArrayType.ToEETypePtr().ArrayElementType);
-                int length = lengths[0];
-                if (length < 0)
-                    throw new OverflowException(); // For compat: we need to throw OverflowException(): Array.CreateInstance throws ArgumentOutOfRangeException()
-                return Array.CreateInstance(Type.GetTypeFromHandle(elementTypeHandle), length);
+                Type elementType = Type.GetTypeFromHandle(new RuntimeTypeHandle(typeHandleForArrayType.ToEETypePtr().ArrayElementType));
+                return RuntimeImports.RhNewArray(elementType.MakeArrayType().TypeHandle.ToEETypePtr(), lengths[0]);
             }
 
             // Create a local copy of the lenghts that cannot be motified by the caller
-            int* pLengths = stackalloc int[lengths.Length];
+            int* pImmutableLengths = stackalloc int[lengths.Length];
```

Comment:
I assume the allowed arity is small enough that lengths here won't result in a stack overflow?

Reply:
Yes, that's correct. We make the similar assumptions in number of other places.